packages/amplify-category-api/src/provider-utils/awscloudformation/PipelineWithAwaiter.ts

```diff
@@ -0,0 +1,380 @@
+import * as codebuild from '@aws-cdk/aws-codebuild';
+import * as codepipeline from '@aws-cdk/aws-codepipeline';
+import * as codepipelineactions from '@aws-cdk/aws-codepipeline-actions';
+import * as ecr from '@aws-cdk/aws-ecr';
+import * as ecs from '@aws-cdk/aws-ecs';
+import * as iam from '@aws-cdk/aws-iam';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as cdk from '@aws-cdk/core';
+import * as custom from '@aws-cdk/custom-resources';
+import fs from 'fs-extra';
+import path from 'path';
+import { DEPLOYMENT_MECHANISM } from './ecs-stack';
+import { getGitHubOwnerRepoFromPath } from './utils/github';
+
+type PipelineAwaiterProps = {
+  pipeline: codepipeline.Pipeline;
+  artifactBucketName?: string;
+  artifactKey?: string;
+  deploymentMechanism: DEPLOYMENT_MECHANISM;
+};
+
+export type GitHubSourceActionInfo = {
+  path: string;
+  tokenSecretArn: string;
+};
+
+class PipelineAwaiter extends cdk.Construct {
+  constructor(scope: cdk.Construct, id: string, props: PipelineAwaiterProps) {
+    const { pipeline, artifactBucketName, artifactKey, deploymentMechanism } = props;
+
+    const { pipelineArn, pipelineName } = pipeline;
+
+    const onEventHandler = new lambda.Function(scope, `${id}CustomEventHandler`, {
+      runtime: lambda.Runtime.NODEJS_12_X,
+      handler: 'index.handler',
+      code: lambda.Code.fromInline(`exports.handler = async function ({ RequestType, PhysicalResourceId, ResourceProperties }) {
+                switch (RequestType) {
+                    case 'Delete':
+                    case 'Update':
+                        return { PhysicalResourceId };
+                }
+
+                const { pipelineName } = ResourceProperties;
+
+                const result = {
+                    PhysicalResourceId: \`pipelineawaiter-\${pipelineName}\`
+                };
+
+                return result;
+            };`),
+      timeout: cdk.Duration.minutes(5),
```

Comment:
Can max it out to 15 maybe - any specific reason for 5 mins?

Reply:
This lambda does practically nothing, it just returns a friendly physical id for the awaiter.

Most of the time it will finish way before the minimum billable time for lambda (100ms)